### PR TITLE
[FEATURE] introduce the snippets into the alignment module as well.

### PR DESCRIPTION
**Pull request by joshuak94**

### CODE REVIEW

**Comment by marehr** [comment]
can you rebase on top of master again?

**Comment by joshuak94** [comment]
@marehr Done!

File: test/snippet/alignment/scoring/aminoacid_scoring_scheme.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+scheme.set_hamming_distance();
+std::cout << "Hamming distance between T and S: " << (int) scheme.get_score(aa27::T, aa20::S) << "\n"; // == -1
+std::cout << "Hamming distance between T and T: " << (int) scheme.get_score(aa27::T, aa20::T) << "\n"; // == 0
+//! [two letters]
+}
+
+{
+//! [edit matrix]
+aminoacid_scoring_scheme scheme;
```
Comment (smehringer) [inline test/snippet/alignment/scoring/aminoacid_scoring_scheme.cpp]: Why did you change from initializing the scheme directly with the matrix 
(`scheme.set_similarity_matrix(aminoacid_similarity_matrix::BLOSUM80);` [here](https://github.com/seqan/seqan3/pull/390/files#diff-6c0ebd7dd2a142616084565cf9b4315eL114) 
to default initialization and the `set_similarity_matrix()` function?

File: test/snippet/alignment/scoring/nucleotide_scoring_scheme.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+{
+
+{
+//! [two letters]
+nucleotide_scoring_scheme scheme; // hamming is default
+std::cout << "Score between DNA A and G: " << (int) scheme.get_score(dna5::A, dna5::G) << "\n"; // == -1
+std::cout << "Score between DNA A and A: " << (int) scheme.get_score(dna5::A, dna5::A) << "\n"; // == 0
+
+scheme.set_simple_scheme(match_score{3}, mismatch_score{-2});
+std::cout << "Score between DNA A and RNA G: " << (int) scheme.get_score(dna5::A, rna15::G) << "\n"; // == -2
```
Comment (smehringer) [inline test/snippet/alignment/scoring/nucleotide_scoring_scheme.cpp]: Maybe add the alphabet size : `Score between DNA5 A and RNA15 G`?

File: test/snippet/alignment/scoring/nucleotide_scoring_scheme.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+std::cout << "Score between DNA A and RNA A: " << (int) scheme.get_score(dna5::A, rna15::A) << "\n"; // == 3
+// you can score differenct nucleotides  ^
+//! [two letters]
+}
+
+{
+//! [edit matrix]
+nucleotide_scoring_scheme scheme; // hamming is default
```
Comment (smehringer) [inline test/snippet/alignment/scoring/nucleotide_scoring_scheme.cpp]: hamming **distance** :nail_care:

File: test/snippet/alignment/scoring/aminoacid_scoring_scheme.cpp (hunk 1 above)
Comment (joshuak94) [reply test/snippet/alignment/scoring/aminoacid_scoring_scheme.cpp]: It's just because the `edit matrix` example is in a different scope from the example above it, where `scheme` was declared. So I have to re-declare the `scheme` variable before I could do anything with it. I think `scheme.set_similarity_matrix(...)` in that context was used to change the similarity matrix from whatever it was before, I don't think it's initialized like that in the example.

File: test/snippet/alignment/scoring/nucleotide_scoring_scheme.cpp (hunk 3 above)
Comment (joshuak94) [reply test/snippet/alignment/scoring/nucleotide_scoring_scheme.cpp]: :+1:

File: test/snippet/alignment/scoring/nucleotide_scoring_scheme.cpp (hunk 2 above)
Comment (joshuak94) [reply test/snippet/alignment/scoring/nucleotide_scoring_scheme.cpp]: :ok:

**Comment by joshuak94** [comment]
```
The following tests FAILED:
	  1 - snippet/alignment/scoring/aminoacid_scoring_scheme_snippet (Not Run)
	  2 - snippet/alignment/scoring/nucleotide_scoring_scheme_snippet (Not Run)
Errors while running CTest
``` Does anyone know why?

**Comment by smehringer** [comment]
You need to look further up for the error.
it says
```
/home/travis/build/seqan/seqan3/test/snippet/alignment/scoring/aminoacid_scoring_scheme.cpp:15:61: error: ‘class seqan3::aminoacid_scoring_scheme<signed char>’ has no member named ‘get_score’; did you mean ‘score’?
 std::cout << "BLOSUM62 score for T and S: " << (int) scheme.get_score(aa27::T, aa27::S) << "\n"; // == 1
                                                             ^~~~~~~~~
                                                             score
```

Check the scoring scheme documentation for the function name

File: test/snippet/alignment/scoring/aminoacid_scoring_scheme.cpp (hunk 1 above)
Comment (smehringer) [reply test/snippet/alignment/scoring/aminoacid_scoring_scheme.cpp]: Yes all true, but still, in the original documentation block (https://github.com/seqan/seqan3/pull/390/files#diff-6c0ebd7dd2a142616084565cf9b4315eL114) it IS initilaized with the matrix not the function, so you might want to change it back.

Comment (joshuak94) [reply test/snippet/alignment/scoring/aminoacid_scoring_scheme.cpp]: Done! :heavy_check_mark:
